android/expoview/src/main/java/host/exp/exponent/notifications/delegates/ScopedExpoPresentationDelegate.kt

```diff
@@ -0,0 +1,45 @@
+package host.exp.exponent.notifications.delegates
+
+import android.content.Context
+import android.os.Parcel
+import android.service.notification.StatusBarNotification
+import android.util.Log
+import expo.modules.notifications.notifications.model.Notification
+import expo.modules.notifications.notifications.model.NotificationBehavior
+import expo.modules.notifications.notifications.model.NotificationRequest
+import expo.modules.notifications.notifications.presentation.builders.ExpoNotificationBuilder
+import expo.modules.notifications.notifications.service.SharedPreferencesNotificationCategoriesStore
+import expo.modules.notifications.service.delegates.ExpoPresentationDelegate
+import host.exp.exponent.notifications.ScopedCategoryAwareNotificationBuilder
+import host.exp.exponent.notifications.model.ScopedNotificationRequest
+import java.util.*
+
+class ScopedExpoPresentationDelegate(context: Context) : ExpoPresentationDelegate(context) {
+  override fun createNotification(notification: Notification, notificationBehavior: NotificationBehavior?): android.app.Notification =
+    ScopedCategoryAwareNotificationBuilder(context, SharedPreferencesNotificationCategoriesStore(context)).also {
+      it.setNotification(notification)
+      it.setAllowedBehavior(notificationBehavior)
+    }.build()
+
+  override fun getNotification(statusBarNotification: StatusBarNotification): Notification? {
+    statusBarNotification.notification.extras.getByteArray(ExpoNotificationBuilder.EXTRAS_MARSHALLED_NOTIFICATION_REQUEST_KEY)?.let {
+      try {
+        with(Parcel.obtain()) {
+          this.unmarshall(it, 0, it.size)
+          this.setDataPosition(0)
+          val request: ScopedNotificationRequest = ScopedNotificationRequest.CREATOR.createFromParcel(this)
+          this.recycle()
```

Comment:
```suggestion
          unmarshall(it, 0, it.size)
          setDataPosition(0)
          val request: ScopedNotificationRequest = ScopedNotificationRequest.CREATOR.createFromParcel(this)
          recycle()
```
`this` keyword is not needed here.